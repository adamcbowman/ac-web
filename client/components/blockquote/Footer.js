import React from 'react'
import PropTypes from 'prop-types'
import styles from './Blockquote.css'

Footer.propTypes = {
    children: PropTypes.node.isRequired,
}

export default function Footer({ children }) {
    return <footer className={styles.Footer}>{children}</footer>
}
