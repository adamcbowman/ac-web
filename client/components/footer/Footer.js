import React from 'react'
import { neverUpdate } from '~/compose'
import {Link} from 'react-router-dom'
import CSSModules from 'react-css-modules'
import styles from './Footer.css'

function Footer() {
    const year = new Date().getFullYear()

    return (
        <footer styleName="Container">
            <div styleName="Content">
                <nav styleName="Nav">
                    <Link styleName="Link" to="/about#contact-us">Contact</Link>
                    <Link styleName="Link" to="/privacy-policy">
                        Privacy Policy
                    </Link>
                    <Link styleName="Link" to="/terms-of-use">
                        Terms of use
                    </Link>
                </nav>
                <span styleName="Rights">
                    Â©{year} Avalanche Canada, All Rights Reserved
                </span>
            </div>
        </footer>
    )
}

export default neverUpdate(CSSModules(Footer, styles))
