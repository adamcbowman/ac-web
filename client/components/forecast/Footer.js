import React from 'react'
import PropTypes from 'prop-types'
import StaticComponent from 'components/StaticComponent'
import Panel, { INVERSE } from 'components/panel'
import RatingExplanation from 'components/forecast/RatingExplanation'
import { Generic } from 'prismic/components'
import ArchiveDatePicker from './ArchiveDatePicker'
import styles from './Forecast.css'

export default class Footer extends StaticComponent {
    static propTypes = {
        children: PropTypes.node,
    }
    get children() {
        return [
            <ArchivedBulletins />,
            <DangerRatings />,
            <Inbox />,
            <Disclaimer />,
        ]
    }
    render() {
        return (
            <footer className={styles.Footer}>
                {this.props.children || this.children}
            </footer>
        )
    }
}

export class ArchivedBulletins extends StaticComponent {
    static propTypes = {
        date: PropTypes.instanceOf(Date),
        region: PropTypes.string.isRequired,
    }
    static defaultProps = {
        date: new Date(),
    }
    render() {
        const { date, region } = this.props

        return (
            <FooterPanel header="Archived bulletins">
                <ArchiveDatePicker date={date} region={region} />
            </FooterPanel>
        )
    }
}

export class DangerRatings extends StaticComponent {
    render() {
        return (
            <FooterPanel header="Danger Ratings Explained">
                <RatingExplanation />
            </FooterPanel>
        )
    }
}

export class Inbox extends StaticComponent {
    render() {
        return (
            <FooterPanel header="Avalanche Forecasts in your Inbox">
                <div className={styles.PanelContent}>
                    <Generic uid="forecast-rss-message" />
                </div>
            </FooterPanel>
        )
    }
}

export class Disclaimer extends StaticComponent {
    render() {
        return (
            <FooterPanel header="Forecast Disclaimer">
                <div className={styles.PanelContent}>
                    <Generic uid="forecast-disclaimer" />
                </div>
            </FooterPanel>
        )
    }
}

// Utils
function FooterPanel({ header, children }) {
    return (
        <Panel theme={INVERSE} expandable header={header}>
            {children}
        </Panel>
    )
}
