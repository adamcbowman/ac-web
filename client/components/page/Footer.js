import {DOM} from 'react'
import {Element} from '~/compose'
import styles from './Page.css'

export default Element({
    name: 'Footer',
    component: DOM.footer,
    styles,
})
