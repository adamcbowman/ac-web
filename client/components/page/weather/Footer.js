import React from 'react'
import PropTypes from 'prop-types'
import styles from './Weather.css'
import { Mailto } from 'components/anchors'

Footer.propTypes = {
    showFeedbackAnchor: PropTypes.bool,
}

export default function Footer({ showFeedbackAnchor = false }) {
    return (
        <footer className={styles.Footer}>
            Forecasts and graphics produced by the Meteorological Service of
            Canada (MSC){' '}
            {showFeedbackAnchor && (
                <Mailto
                    email="ec.cpip-mwf-pspc.ec@canada.ca"
                    subject="Mountain Weather Forecast Feedback">
                    Send feedback to MSC
                </Mailto>
            )}
        </footer>
    )
}
