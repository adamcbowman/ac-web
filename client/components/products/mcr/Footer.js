import React, { Component } from 'react'
import styles from './MountainConditionsReport.css'
import logo from 'styles/mcr-logo.jpg'

export default class Footer extends Component {
    shouldComponentUpdate() {
        return false
    }
    render() {
        const url = 'https://mountainconditions.com'

        return (
            <div className={styles.Footer}>
                <a href={url} title="Mountain Conditions Reports" target="mcr">
                    <img src={logo} alt="MCR" />
                </a>
                <div>
                    For more photos and more information about the professional
                    mountain guide who created this report visit:
                    <a href={url} target="mcr">
                        mountainconditions.com
                    </a>
                </div>
            </div>
        )
    }
}
