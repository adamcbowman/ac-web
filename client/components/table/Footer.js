import React from 'react'
import PropTypes from 'prop-types'
import styles from './Table.css'

Footer.propTypes = {
    children: PropTypes.node.isRequired,
}

export default function Footer({ children }) {
    return <tfoot className={styles.Footer}>{children}</tfoot>
}
