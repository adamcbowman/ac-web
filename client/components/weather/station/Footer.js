import React from 'react'
import PropTypes from 'prop-types'
import { Generic } from 'prismic/components'

Footer.propTypes = {
    children: PropTypes.node.isRequired,
}

export default function Footer({ children }) {
    return (
        <footer>
            <Generic uid="weather-station-disclaimer" />
            {children}
        </footer>
    )
}
