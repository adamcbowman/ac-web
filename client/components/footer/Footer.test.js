import React from 'react'
import Renderer from 'react-test-renderer'
import Footer from './'

test('footer component', () => {
    const footer = Renderer.create(<Footer />)

    expect(footer).toMatchSnapshot()
})
