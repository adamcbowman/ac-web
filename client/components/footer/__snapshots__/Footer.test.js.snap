// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer component 1`] = `
<footer
  className="Container"
>
  <div
    className="Content"
  >
    <nav
      className="Nav"
    >
      <a
        className="Link"
        href="/about#contact-us"
        onClick={[Function]}
      >
        Contact
      </a>
      <a
        className="Link"
        href="/privacy-policy"
        onClick={[Function]}
      >
        Privacy Policy
      </a>
      <a
        className="Link"
        href="/terms-of-use"
        onClick={[Function]}
      >
        Terms of use
      </a>
    </nav>
    <span
      className="Rights"
    >
      Â©
      2019
       Avalanche Canada, All Rights Reserved
    </span>
  </div>
</footer>
`;
